### ISSUE TITLE
OpenStf

### ISSUE THREAD

**Issue by sh0kunin** [dimasaryo/stf-k8s#3]
How do you access OpenStf in kubernetes?
Can you please explain how the stf-app and stf-processor working in kubernetes - Some screenshots will be helpful?

**Comment by dimasaryo** [comment]
So, all services inside [stf-services](https://github.com/dimasaryo/stf-k8s/tree/master/stf-services) will running in the kubernetes cluster. They're connected using internal ip. Only `stf-triproxy-dev` will have external ip because later we will connect our `stf-provider` service from local machine to `stf-triproxy-dev` in the kubernetes. Then all of those services will be wrapped using `stf-nginx`. User can access the openstf from internal/external ip of the `stf-nginx`.

**Comment by sh0kunin** [comment]
There are no service for `stf-provider` and `stf-reaper` is that intentional or we missing something?

**Comment by dimasaryo** [comment]
yep, for `stf-provider` that is intentional because we need to run it on local machine where the device is connected. but for stf-reaper you can find it here [stf-reaper](https://github.com/dimasaryo/stf-k8s/tree/master/stf-services/stf-reaper)

**Comment by dimasaryo** [comment]
Here is for [stf-provider](https://github.com/dimasaryo/stf-k8s#stf-provider)

**Comment by sh0kunin** [comment]
For stf-reaper it a replication controller no service required?

**Comment by dimasaryo** [comment]
ah, correct. I miss understanding about the `service`.yeah,  stf-reaper and stf-processor only a replication controller. They don't need a service because they only push/subscribe from stf-triproxy.

**Comment by sh0kunin** [comment]
Aah, alright. And how do you access the OpenStf dashboard?

**Comment by dimasaryo** [comment]
Just open the ip of stf-nginx. The stf-nginx service is a load balancer, so you should get an external ip. Here is more reference for [external load balancer type](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/)

**Comment by sh0kunin** [comment]
how does your stf-app gets connected to rethinkdb-proxy, I see it is always Connecting but not connected to rethinkdb-proxy?

**Comment by dimasaryo** [comment]
Have you test it by running `stf-migrate`? or if you follow the example, try to remove the rethinkdb-admin pod and service. Last time I had an issue with that. somehow rethinkdb-rc and admin have their own storage. If still not working, try to replace `"tcp://rethinkdb-driver:28015"` and `"tcp://rethinkdb-proxy:28015"` using its ip. If the `stf-migrate` success, the `stf-app` should also working properly.

**Comment by 45minutepromise** [comment]
Yeah with migrate it works. Thanks
Also, do you think an ingress can be replaces with nginx to listen on different ports or we need nginx here?

**Comment by sh0kunin** [comment]
@dimasaryo in your stf-provider what do you expect to be `tcp://<TRIPROXY DEV HERE>:7270`, `<TRIPROXY DEV HERE>`? is it not `tcp://triproxy-dev-here:7270`

**Comment by dimasaryo** [comment]
No, since the stf-provider is executed in the local machine, it should be the external ip/url of stf-triproxy-dev service. You can get it using command:
```
# kubectl get service stf-triproxy-dev
```
